### ISSUE TITLE
[BUG] fd-avatar-group focus, overflow bugs

### ISSUE THREAD

**Issue by egorwow70** [SAP/fundamental-ngx#12369]
#### Is this a bug, enhancement, or feature request?

Couple of bugs connected to fd-avatar-group component from fundamental-ngx/core library.

Control - https://docs-0-49--fundamental-ngx.netlify.app/#/core/avatar-group.

#### Describe your proposal.

I think it should be fixed on the library side.

#### Can you handle that on the application side

No.

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)

- "@fundamental-ngx/core": "0.49.0",
- "@sap-theming/theming-base-content": "11.13.0",
- "fundamental-styles": "0.34.0",
- "@angular/core": "17.3.0",

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;

**a) Bug with focus:**

1. Open stackblitz example - https://stackblitz.com/edit/uy2qbh?file=src%2Fapp%2Fdynamic-page-tabs-example.component.html
2. Open default tabs example.
3. When there is more than one avatar group control on the page, firstly click on the first one, you should see a blue border, that highlights a focus state.
4. After click on the second control, focus will still remain on the first one.
![image](https://github.com/user-attachments/assets/f7d645c1-d59f-47e6-88c0-125653f96ad1)

**b) Bug with switching tabs and avatar group control rendering:**

1. Open stackblitz example - https://stackblitz.com/edit/uy2qbh?file=src%2Fapp%2Fdynamic-page-tabs-example.component.html
2. Open default tabs example.
3. Select the second tab.
4. Go back to the first tab.
5. You see only "+n more button", but you should see the whole avatar group row with images as well.
![image](https://github.com/user-attachments/assets/3a49a44f-bcce-41a9-b6c4-651dc938232f)

**c) Not a bug, but a feature request**

Is it possible to hide a standard avatar group item details popover header? But then we need access to navigate between group and item views
![image](https://github.com/user-attachments/assets/457849a5-0d14-4963-8240-83b90e636397)

I found the way how to custom the body of the popover with <ng-template #details>, but didn't find the way how to hide this header
![image](https://github.com/user-attachments/assets/7e29993e-6fd8-4bb3-98d8-54031226db0b)

**d) Not a bug, but a feature request**
 Is it possible to open group popover from the center of the avatar group?
 
 Like this:
 
![image](https://github.com/user-attachments/assets/fec09f04-7782-4a66-be49-d551cfe12890)

**e) Not a bug, but a feature request**

Is it possible to make +n button in circle shape, because avatars are mostly used with circle type? Maybe via configuration prop is the best choice for it

design:
![image](https://github.com/user-attachments/assets/70fe8be7-c03b-43b1-8427-951896ae4973)

**f) Not a bug, but a feature request**

Is it possible to have a property to set the number of visible circles? we have a strict requirement to display 5 avatars and after +n button

design:
![image](https://github.com/user-attachments/assets/70fe8be7-c03b-43b1-8427-951896ae4973)

#### Please provide stackblitz example(s).

Link - https://stackblitz.com/edit/uy2qbh?file=src%2Fapp%2Fdynamic-page-tabs-example.component.html.

#### Did you check the documentation and the API?

Yes.

#### Did you search for similar issues?

Yes.

**Comment by egorwow70** [comment]
@khotcholava  @droshev 

As for point c) I found the way how to customize overflow popover inside fd-avatar-group, but have a question, is it possible to close this popover? We need to display a cross icon inside the popover, that closes it

![image](https://github.com/user-attachments/assets/957b5275-d0fb-4a65-b55b-a917c77c0f70)

**Comment by khotcholava** [comment]
@droshev

1. Please refer to point A. I've made the necessary adjustments, but when I click outside the popover, the focus on the selected popover disappears. Should it disappear, or should it remain on the selected avatar group?

2. FYI:  Point **C** is a new feature as well as **D,E,F**  points.

**Comment by droshev** [comment]
@khotcholava where can i see your PR?

**Comment by droshev** [comment]
@egorwow70 regarding point C you can use `AvatarGroupOverflowBodyDirective` and pass any template you want. [Example](https://stackblitz.com/edit/sakxse?file=src%2Fmain.ts)

@khotcholava I guess you can open PR for the others:
point A and B are bugs and should be fixes. C should be doable. D is a BUG. For E and F we should offer that through the API

**Comment by egorwow70** [comment]
@droshev 

Hello, Deno. Our team is really interested in those fixes, I see that you attached a November 2024 label, do you know when those fixes will be released?

**Comment by droshev** [comment]
@egorwow70 @khotcholava is working on that ticket and we should be able to fix it anytime now.

**Comment by khotcholava** [comment]
Imagine this scenario: I have a group of multiple avatars. When I click on the first avatar, its associated popover opens. If I then click on the second avatar and close its popover, the focus stays on the first avatar. Is this the expected behavior?
For more details, refer to section A regarding this issue. @droshev

**Comment by droshev** [comment]
@nikolay-kolarov Can you advise us in this question?

**Comment by egorwow70** [comment]
@droshev 

Hello, Deno, any news regarding this ticket?

**Comment by egorwow70** [comment]
@droshev 

Hello, Deno. Any news regarding this issue?

**Comment by egorwow70** [comment]
@droshev 
We are on 0.54.1

**Comment by droshev** [comment]
@egorwow70 do you think you can ping me in Teams?

**Comment by MariaIDineva** [comment]
Hello, @egorwow70 !

Here's a short report in relation to your issue. There are also a couple of questions to you, which we need your input for before we can proceed:

### Bug a) Focus ring persists on first avatar group when clicking the second

**Status: not reproducible in v0.65.0-rc.11.**

### Bug b) After switching tabs, only "+N more" shows — avatars are missing

**Status: not reproducible as originally reported in v0.65.0-rc.11**, because the default `fdp-icon-tab-bar` (non-stacked mode) destroys inactive tab content via `@if`, so the avatar group is recreated from scratch on each tab switch and never enters the broken state.

However, investigation uncovered a **latent bug** that manifests whenever an `fd-avatar-group` is hidden by a CSS `display:none` ancestor rather than being destroyed — for example, a custom tab implementation that hides content with CSS instead of `@if`.

A [PR with this fix](https://github.com/SAP/fundamental-ngx/pull/14539) has been raised against `main`.

### Feature c) Hide a standard avatar group item details popover header

A [separate issue](https://github.com/SAP/fundamental-ngx/issues/14545) has been logged to track this feature request separately as there are some open questions there, which are not covered by the designs we have. We would appreciate your input with details about the use case that you have as well as how we could possibly preserve navigation if we remove the header.

### Feature d) Open group overflow popover from the center of the avatar group

**Status: implemented [here](https://github.com/SAP/fundamental-ngx/pull/14546).**

**Change:** a new `popoverPlacement` signal input on `AvatarGroupComponent` (only applies to `type="group"`),
defaulting to `'bottom'` so the popover opens centered out of the box. Consumers who need the previous
left-aligned behaviour can pass `[popoverPlacement]="'bottom-start'"`.

### Feature e) Circle-shaped overflow button

**Status: implemented [here](https://github.com/SAP/fundamental-ngx/pull/14546).**

**Change:** added `readonly overflowButtonShape = input<'circle' | 'square'>('square')` to `AvatarGroupComponent`.
The template binds `[circle]="overflowButtonShape() === 'circle'"` on both overflow button usages (individual type
and group type). Passing `overflowButtonShape="circle"` makes the overflow button match the circular avatars.

### Feature f) Cap visible avatars at a fixed count

**Status: implemented [here](https://github.com/SAP/fundamental-ngx/pull/14546).**

**Change:** added `readonly maxVisibleItems = input<number | null>(null)` to `AvatarGroupComponent` and a corresponding
`@Input() maxVisibleItems: number | null = null` to `AvatarGroupHostComponent`. 

**Behaviour:** wide container → shows exactly up to N avatars (count is the limiting factor). Narrow container →
width constraint kicks in first within the cap; items move to overflow as the container shrinks. `forceVisibility`
items are always shown and reduce the number of regular slots available.


## Questions

1. Can you please add some more information regarding `Feature C` in the [issue](https://github.com/SAP/fundamental-ngx/issues/14545) that we filed. We are interested to know more about your use case and what your proposition is regarding the missing navigation button when the header is hidden.
2. Can you please confirm that `Feature F`, the way it is implemented now, is what you desire - a `maxVisibleItems` items that would not allow display of more than the desired number of avatars but it would take into account container width and move items into the overflow in case the width shrinks?
